valetudo-map-card.js

```diff
@@ -617,47 +547,93 @@ class ValetudoMapCard extends HTMLElement {
 
     // Set card title and hide the header completely if the title is set to an empty value
     if (!this._config.title) {
-        this.cardHeader.style.display = 'none';
+      this.cardHeader.style.display = 'none';
     } else {
-        this.cardHeader.style.display = 'block';
-    };
+      this.cardHeader.style.display = 'block';
+    }
     this.cardTitle.textContent = this._config.title;
 
     // Set container card background color
     if (this._config.background_color) {
       this.cardContainer.style.background = this._config.background_color;
     } else {
       this.cardContainer.style.background = null;
-    };
+    }
 
     if (!Array.isArray(this._config.custom_buttons)) {
       this._config.custom_buttons = [];
-    };
+    }
   };
 
   set hass(hass) {
     // Home Assistant 0.110.0 may call this function with undefined sometimes if inside another card
     if (hass === undefined) return;
 
     this._hass = hass;
-    const config = this._config;
+
     let mapEntity = this._hass.states[this._config.entity];
+
+    let attributes = mapEntity ? mapEntity.attributes : undefined;
+
+
+    if (mapEntity && mapEntity['state'] !== 'unavailable' && attributes && attributes["entity_picture"]) {
+      if (new Date().getTime() - 30000 > this.lastMapPoll.getTime()) {
```

Comment:
Instead of polling every 30 seconds, wouldn't it be better to poll like every second but keep the last value in another variable and only call handleDrawing if the new value changed?

Reply:
That would mean > 120kb of map/image download every second

Reply:
Maybe we should require also giving the MQTT vacuum element then and base how often we refresh on if the vacuum is cleaning or docked? https://github.com/TheLastProject/lovelace-valetudo-map-card/blob/master/valetudo-map-card.js#L806-L810

Reply:
Good idea!